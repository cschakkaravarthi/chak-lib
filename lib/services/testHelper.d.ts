export declare const shallowRender: (toRender: any) => Record<string, any>;
//# sourceMappingURL=testHelper.d.ts.map