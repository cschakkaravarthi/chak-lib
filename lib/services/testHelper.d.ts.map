{"version":3,"file":"testHelper.d.ts","sourceRoot":"","sources":["../../src/services/testHelper.tsx"],"names":[],"mappings":"AAKA,eAAO,MAAM,aAAa,wCAIzB,CAAC"}