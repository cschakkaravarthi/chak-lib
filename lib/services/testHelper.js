"use strict";
/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var shallow_1 = __importDefault(require("react-test-renderer/shallow"));
exports.shallowRender = function (toRender) {
    var renderer = shallow_1["default"].createRenderer();
    renderer.render(toRender);
    return renderer.getRenderOutput();
};
//# sourceMappingURL=testHelper.js.map