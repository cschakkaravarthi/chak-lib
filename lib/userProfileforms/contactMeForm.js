"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var react_bootstrap_1 = require("react-bootstrap");
var textbox_1 = require("../components/textbox");
var contentWrapper_1 = require("../components/contentWrapper");
var primaryButton_1 = require("../components/primaryButton");
var cancelButton_1 = require("../components/cancelButton");
exports.ContactMeForm = function (props) {
    var title = props.title, hideMobileNumberOnProfile = props.hideMobileNumberOnProfile, mobileNumber = props.mobileNumber, handleCheckboxWithIdentifier = props.handleCheckboxWithIdentifier, handleSave = props.handleSave, handleCancel = props.handleCancel, handleChange = props.handleChange, mobileNumberMaxLength = props.mobileNumberMaxLength;
    return (react_1["default"].createElement(contentWrapper_1.ContentWrapper, { title: title, subtitle: "All information is visible to anyone on UMG Central.", subtitleClass: "gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle", titleClass: "font-size-md font-weight-bold text-uppercase d-lg-inline flex-row mb-0", topBorder: 'darkTurquoise', wrapperClass: 'h-100' },
        react_1["default"].createElement(react_bootstrap_1.Form, { className: 'form-user-description mt-4' },
            react_1["default"].createElement(react_bootstrap_1.Form.Row, null,
                react_1["default"].createElement(react_bootstrap_1.Col, null,
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formMobile", className: "pr-md-2" },
                        react_1["default"].createElement(textbox_1.Textbox, { controlId: "formMobile", className: "font-weight-bold text-condensed", maxLength: mobileNumberMaxLength, name: "mobileNumber", label: "Mobile", placeholder: "+1 999 999 9999", type: "tel", value: mobileNumber, handleChange: handleChange, allowCheckbox: true, checkboxValue: hideMobileNumberOnProfile, handleCheckbox: function (e) { return handleCheckboxWithIdentifier && handleCheckboxWithIdentifier(e, 'hideMobileNumberOnProfile'); } })))),
            react_1["default"].createElement(react_bootstrap_1.Form.Row, { className: 'mt-2 mb-2' },
                react_1["default"].createElement(react_bootstrap_1.Col, { md: 12, className: 'd-flex justify-content-end mt-2' },
                    react_1["default"].createElement(cancelButton_1.CancelButton, { title: "Cancel", className: 'cancel-button text-underlined text-upper border border-0 mr-3', uniqueIdentifier: "ok", handleClick: handleCancel }),
                    react_1["default"].createElement(primaryButton_1.PrimaryButton, { title: "Save", className: 'text-upper', uniqueIdentifier: "ok", handleClick: handleSave }))))));
};
//# sourceMappingURL=contactMeForm.js.map