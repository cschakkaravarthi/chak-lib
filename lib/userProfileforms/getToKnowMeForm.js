"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var react_bootstrap_1 = require("react-bootstrap");
var textbox_1 = require("../components/textbox");
var textarea_1 = require("../components/textarea");
var primaryButton_1 = require("../components/primaryButton");
var cancelButton_1 = require("../components/cancelButton");
var contentWrapper_1 = require("../components/contentWrapper");
var customCheckbox_1 = require("../components/customCheckbox");
require("../../scss/_forms.scss");
exports.GetToKnowMeForm = function (props) {
    var title = props.title, nickname = props.nickname, languagesSpoken = props.languagesSpoken, professionalSkills = props.professionalSkills, currentlyWorking = props.currentlyWorking, aboutMe = props.aboutMe, hideBirthdayOnProfile = props.hideBirthdayOnProfile, hideAnniversaryOnProfile = props.hideAnniversaryOnProfile, handleSave = props.handleSave, handleCancel = props.handleCancel, handleChange = props.handleChange, birthDay = props.birthDay, birthMonth = props.birthMonth, anniversaryDay = props.anniversaryDay, anniversaryMonth = props.anniversaryMonth, handleHideBirthday = props.handleHideBirthday, handleHideAnniversary = props.handleHideAnniversary, textBoxMaxLength = props.textBoxMaxLength, textArearBoxMaxLength = props.textArearBoxMaxLength;
    var dates = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'];
    var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
    var deafultOption = react_1["default"].createElement("option", { value: "", key: "", selected: true }, "N/A");
    return (react_1["default"].createElement(contentWrapper_1.ContentWrapper, { title: title, titleClass: "text-upper d-lg-inline flex-row font-size-md mb-0", subtitle: "All information is visible to anyone on UMG Central.", subtitleClass: "gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle", topBorder: 'darkTurquoise', wrapperClass: 'h-100' },
        react_1["default"].createElement(react_bootstrap_1.Form, { className: "font-size-sm w-100 form-user-description mt-4" },
            react_1["default"].createElement(react_bootstrap_1.Form.Row, { className: "d-flex justify-content-between" },
                react_1["default"].createElement(react_bootstrap_1.Col, null,
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formNickname", className: "pr-md-2" },
                        react_1["default"].createElement(textbox_1.Textbox, { controlId: "nickname", name: "nickname", maxLength: textBoxMaxLength, className: "font-weight-bold text-condensed font-size-base", label: "Nickname", type: "text", value: nickname, allowCheckbox: false, handleChange: handleChange })))),
            react_1["default"].createElement(react_bootstrap_1.Form.Row, null,
                react_1["default"].createElement(react_bootstrap_1.Col, { lg: 3, md: 7 },
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formBirthMonth" },
                        react_1["default"].createElement(react_bootstrap_1.Form.Label, { className: "font-weight-bold text-condensed font-size-base" }, "Birthday"),
                        react_1["default"].createElement(react_bootstrap_1.Form.Control, { name: "birthMonth", as: "select", value: birthMonth, onChange: handleChange },
                            deafultOption,
                            months.map(function (month) {
                                return react_1["default"].createElement("option", { value: month, key: month }, month);
                            })))),
                react_1["default"].createElement(react_bootstrap_1.Col, { lg: 3, md: 7, className: "paddingRight" },
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formBirthday", className: "day-dropdown" },
                        react_1["default"].createElement(customCheckbox_1.CustomCheckbox, { handleClick: handleHideBirthday, checked: hideBirthdayOnProfile, title: "Hide on profile:", className: "mb-1" }),
                        react_1["default"].createElement(react_bootstrap_1.Form.Control, { name: "birthday", as: "select", value: birthDay, onChange: handleChange },
                            deafultOption,
                            dates.map(function (date) {
                                return react_1["default"].createElement("option", { value: date, key: date }, date);
                            })))),
                react_1["default"].createElement(react_bootstrap_1.Col, { lg: 3, md: 7 },
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formAnniversayMonth" },
                        react_1["default"].createElement(react_bootstrap_1.Form.Label, { className: "font-weight-bold text-condensed font-size-base" }, "Work Anniversary"),
                        react_1["default"].createElement(react_bootstrap_1.Form.Control, { name: "anniversaryMonth", as: "select", value: anniversaryMonth, onChange: handleChange },
                            deafultOption,
                            months.map(function (month) {
                                return react_1["default"].createElement("option", { value: month, key: month }, month);
                            })))),
                react_1["default"].createElement(react_bootstrap_1.Col, { lg: 3, md: 5 },
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formAnniversayday", className: "day-dropdown" },
                        react_1["default"].createElement(customCheckbox_1.CustomCheckbox, { handleClick: handleHideAnniversary, checked: hideAnniversaryOnProfile, title: "Hide on profile:", className: "mb-1" }),
                        react_1["default"].createElement(react_bootstrap_1.Form.Control, { name: "anniversaryDay", as: "select", value: anniversaryDay, onChange: handleChange },
                            deafultOption,
                            dates.map(function (date) {
                                return react_1["default"].createElement("option", { value: date, key: date }, date);
                            }))))),
            react_1["default"].createElement(react_bootstrap_1.Form.Row, null,
                react_1["default"].createElement(react_bootstrap_1.Col, null,
                    react_1["default"].createElement(react_bootstrap_1.Form.Group, { controlId: "formLanguagesSpoken", className: "pr-md-2" },
                        react_1["default"].createElement(textbox_1.Textbox, { controlId: "formLanguagesSpoken", className: "font-weight-bold text-condensed font-size-base", name: 'languagesSpoken', maxLength: textBoxMaxLength, label: "Languages Spoken", type: "text", value: languagesSpoken, allowCheckbox: false, handleChange: handleChange })))),
            react_1["default"].createElement(textarea_1.Textarea, { controlId: "formSkills", className: "font-weight-bold text-condensed font-size-base", maxLength: textArearBoxMaxLength, name: "professionalSkills", label: "Professional Skills", value: professionalSkills, rows: 3, handleChange: handleChange }),
            react_1["default"].createElement(textarea_1.Textarea, { controlId: "formCurrentlyWorking", className: "font-weight-bold text-condensed font-size-base", maxLength: textArearBoxMaxLength, name: "currentlyWorking", label: "What I'm currently working on", value: currentlyWorking, rows: 3, handleChange: handleChange }),
            react_1["default"].createElement(textarea_1.Textarea, { controlId: "formAboutme", className: "font-weight-bold text-condensed font-size-base", maxLength: textArearBoxMaxLength, name: "aboutMe", label: "Something about me", value: aboutMe, rows: 3, handleChange: handleChange }),
            react_1["default"].createElement(react_bootstrap_1.Form.Row, { className: 'mt-2 mb-2' },
                react_1["default"].createElement(react_bootstrap_1.Col, { md: 12, className: 'd-flex justify-content-end mt-2' },
                    react_1["default"].createElement(cancelButton_1.CancelButton, { title: "Cancel", className: 'cancel-button text-underlined text-upper border border-0 mr-3', uniqueIdentifier: "ok", handleClick: handleCancel }),
                    react_1["default"].createElement(primaryButton_1.PrimaryButton, { title: "Save", className: 'text-upper', uniqueIdentifier: "ok", handleClick: handleSave }))))));
};
//# sourceMappingURL=getToKnowMeForm.js.map