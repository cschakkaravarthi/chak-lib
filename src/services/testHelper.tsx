/* eslint @typescript-eslint/no-explicit-any: "off" */
// In this one file we do actually want to use the "any" type. This is a very generic function.

import ShallowRenderer from 'react-test-renderer/shallow';

export const shallowRender = (toRender: any): Record<string, any> => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(toRender);
  return renderer.getRenderOutput();
};
