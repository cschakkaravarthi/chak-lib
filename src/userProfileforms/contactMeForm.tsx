import React, { FC } from 'react';
import { Form, Col } from 'react-bootstrap';
import { Textbox } from '../components/textbox';
import { ContactMeEditCardProps } from '../interfaces';
import { ContentWrapper } from '../components/contentWrapper';
import { PrimaryButton } from '../components/primaryButton';
import { CancelButton } from '../components/cancelButton';

export const ContactMeForm: FC<ContactMeEditCardProps> = props => {
  const {
    title,
    mobileNumber,
    handleSave,
    handleCancel,
    handleChange,
    mobileNumberMaxLength
  } = props;

  return (
    <ContentWrapper title={title} subtitle="All information is visible to anyone on UMG Central." subtitleClass="gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle" titleClass="font-size-md font-weight-bold text-uppercase d-lg-inline flex-row mb-0" topBorder='darkTurquoise' wrapperClass='h-100'>
      <Form className='form-user-description mt-4'>
        <Form.Row>
          <Col>
            <Form.Group controlId="formMobile" className="pr-md-2">
              <Textbox
                controlId="formMobile"
                className="font-weight-bold text-condensed"
                maxLength={mobileNumberMaxLength}
                name="mobileNumber"
                label="Mobile"
                placeholder="+1 999 999 9999"
                type="tel"
                value={mobileNumber}
                handleChange={handleChange}
                allowCheckbox={false}
              />
            </Form.Group>
          </Col>
        </Form.Row>
        <Form.Row className='mt-2 mb-2'>
          <Col
            md={12}
            className='d-flex justify-content-end mt-2'
          >
            <CancelButton
              title="Cancel"
              className='cancel-button text-underlined text-upper border border-0 mr-3'
              uniqueIdentifier="ok"
              handleClick={handleCancel}
            />
            <PrimaryButton
              title="Save"
              className='text-upper'
              uniqueIdentifier="ok"
              handleClick={handleSave}
            />
          </Col>
        </Form.Row>
      </Form>
    </ContentWrapper>
  );
};
