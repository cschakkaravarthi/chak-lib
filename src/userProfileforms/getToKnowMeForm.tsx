import React, { FC } from 'react';
import { Form, Col } from 'react-bootstrap';
import { GetToKnowMeCardProps } from '../interfaces';
import { Textbox } from '../components/textbox';
import { Textarea } from '../components/textarea';
import { PrimaryButton } from '../components/primaryButton';
import { CancelButton } from '../components/cancelButton';
import { ContentWrapper } from '../components/contentWrapper';
import '../../scss/_forms.scss';

export interface GetToKnowMeFormProps extends GetToKnowMeCardProps {
  handleCheckboxWithIdentifier?: (e: React.SyntheticEvent, uniqueIdentifier: string) => void;
  handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  handleSave: (e: React.SyntheticEvent) => void;
  handleCancel: (e: React.SyntheticEvent) => void;
  textBoxMaxLength?: number;
  textArearBoxMaxLength?: number;
}

export const GetToKnowMeForm: FC<GetToKnowMeFormProps> = props => {
  const {
    title,
    nickname,
    languagesSpoken,
    professionalSkills,
    currentlyWorking,
    aboutMe,
    handleSave,
    handleCancel,
    handleChange,
    birthDay,
    birthMonth,
    anniversaryDay,
    anniversaryMonth,
    textBoxMaxLength,
    textArearBoxMaxLength
  } = props;

  const dates = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31'];
  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
  const deafultOption = <option value="" key="" selected>N/A</option>;
  return (
    <ContentWrapper title={title} titleClass="text-upper d-lg-inline flex-row font-size-md mb-0" subtitle="All information is visible to anyone on UMG Central." subtitleClass="gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle" topBorder='darkTurquoise' wrapperClass='h-100'>
      <Form className="font-size-sm w-100 form-user-description mt-4">
        <Form.Row className="d-flex justify-content-between">
          <Col >
            <Form.Group controlId="formNickname" className="pr-md-2">
              <Textbox
                controlId="nickname"
                name="nickname"
                maxLength={textBoxMaxLength}
                className="font-weight-bold text-condensed font-size-base"
                label="Nickname"
                type="text"
                value={nickname}
                allowCheckbox={false}
                handleChange={handleChange}
              />
            </Form.Group>
          </Col>
        </Form.Row>
        <Form.Row>
          <Col lg={3} md={7}>
            <Form.Group controlId="formBirthMonth">
              <Form.Label className="font-weight-bold text-condensed font-size-base">Birthday</Form.Label>
              <Form.Control name="birthMonth" as="select" value={birthMonth} onChange={handleChange}>
                {deafultOption}
                { months.map((month) => {
                  return <option value={month} key={month}>{month}</option>;
                }) }
              </Form.Control>
            </Form.Group>
          </Col>
          <Col lg={3} md={7} className="paddingRight">
            <Form.Group controlId="formBirthday" className="day-dropdown">
              <Form.Control name="birthday" as="select" value={birthDay} onChange={handleChange}>
                {deafultOption}
                { dates.map((date) => {
                  return <option value={date} key={date}>{date}</option>;
                }) }
              </Form.Control>
            </Form.Group>
          </Col>
          <Col lg={3} md={7}>
            <Form.Group controlId="formAnniversayMonth">
              <Form.Label className="font-weight-bold text-condensed font-size-base">Work Anniversary</Form.Label>
              <Form.Control name="anniversaryMonth" as="select" value={anniversaryMonth} onChange={handleChange}>
                {deafultOption}
                { months.map((month) => {
                  return <option value={month} key={month}>{month}</option>;
                }) }
              </Form.Control>
            </Form.Group>
          </Col>
          <Col lg={3} md={5}>
            <Form.Group controlId="formAnniversayday" className="day-dropdown">
              <Form.Control name="anniversaryDay" as="select" value={anniversaryDay} onChange={handleChange}>
                {deafultOption}
                { dates.map((date) => {
                  return <option value={date} key={date}>{date}</option>;
                }) }
              </Form.Control>
            </Form.Group>
          </Col>
        </Form.Row>

        <Form.Row>
          <Col>
            <Form.Group controlId="formLanguagesSpoken" className="pr-md-2">
              <Textbox
                controlId="formLanguagesSpoken"
                className="font-weight-bold text-condensed font-size-base"
                name="languagesSpoken"
                maxLength={textBoxMaxLength}
                label="Languages Spoken"
                type="text"
                value={languagesSpoken}
                allowCheckbox={false}
                handleChange={handleChange}
              />
            </Form.Group>
          </Col>
        </Form.Row>

        <Textarea
          controlId="formSkills"
          className="font-weight-bold text-condensed font-size-base"
          maxLength={textArearBoxMaxLength}
          name="professionalSkills"
          label="Professional Skills"
          value={professionalSkills}
          rows={3}
          handleChange={handleChange}
        />

        <Textarea
          controlId="formCurrentlyWorking"
          className="font-weight-bold text-condensed font-size-base"
          maxLength={textArearBoxMaxLength}
          name="currentlyWorking"
          label="What I'm currently working on"
          value={currentlyWorking}
          rows={3}
          handleChange={handleChange}
        />

        <Textarea
          controlId="formAboutme"
          className="font-weight-bold text-condensed font-size-base"
          maxLength={textArearBoxMaxLength}
          name="aboutMe"
          label="Something about me"
          value={aboutMe}
          rows={3}
          handleChange={handleChange}
        />
        <Form.Row className='mt-2 mb-2'>
          <Col
            md={12}
            className='d-flex justify-content-end mt-2'
          >
            <CancelButton
              title="Cancel"
              className='cancel-button text-underlined text-upper border border-0 mr-3'
              uniqueIdentifier="ok"
              handleClick={handleCancel}
            />

            <PrimaryButton
              title="Save"
              className='text-upper'
              uniqueIdentifier="ok"
              handleClick={handleSave}
            />
          </Col>
        </Form.Row>
      </Form>
    </ContentWrapper>
  );
};
