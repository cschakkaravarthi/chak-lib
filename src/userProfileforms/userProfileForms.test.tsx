import React from 'react';
import { shallowRender } from '../services/testHelper';
import { GetToKnowMeForm } from './getToKnowMeForm';
import { ContactMeForm } from './contactMeForm';

describe('GetToKnowMeForm', () => {
  it('should render correctly', () => {
    const tree = shallowRender(
      <GetToKnowMeForm
        title="Get to know me"
        nickname="Janie"
        birthDay="1"
        birthMonth="January"
        anniversaryMonth="December"
        anniversaryDay="11"
        languagesSpoken="English, Spanish"
        professionalSkills="Lorem Ipsum is simply dummy text of the printing and typesetting     industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries."
        currentlyWorking="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        aboutMe="Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
        handleCheckboxWithIdentifier={(e: React.SyntheticEvent, uniqueIdentifier: string) => {
          e.preventDefault();
          alert(`Checkbox of ${uniqueIdentifier} is clicked.`);
        }}
        handleSave={() => alert('Save Button clicked')}
        handleCancel={() => alert('Cancel Button clicked')}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const tree = shallowRender(
      <ContactMeForm
        title="Contact Me"
        mobileNumber="9900990099"
        handleSave={() => alert('Save Button clicked')}
        handleCancel={() => alert('Cancel Button clicked')}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
