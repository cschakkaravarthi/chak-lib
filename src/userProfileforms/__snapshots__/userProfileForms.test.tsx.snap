// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetToKnowMeForm should render correctly 1`] = `
<Unknown
  subtitle="All information is visible to anyone on UMG Central."
  subtitleClass="gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle"
  title="Get to know me"
  titleClass="text-upper d-lg-inline flex-row font-size-md mb-0"
  topBorder="darkTurquoise"
  wrapperClass="h-100"
>
  <ForwardRef
    className="font-size-sm w-100 form-user-description mt-4"
    inline={false}
  >
    <ForwardRef
      className="d-flex justify-content-between"
    >
      <ForwardRef>
        <ForwardRef
          className="pr-md-2"
          controlId="formNickname"
        >
          <Unknown
            allowCheckbox={false}
            className="font-weight-bold text-condensed font-size-base"
            controlId="nickname"
            label="Nickname"
            name="nickname"
            type="text"
            value="Janie"
          />
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <ForwardRef>
      <ForwardRef
        lg={3}
        md={7}
      >
        <ForwardRef
          controlId="formBirthMonth"
        >
          <ForwardRef
            className="font-weight-bold text-condensed font-size-base"
            column={false}
            srOnly={false}
          >
            Birthday
          </ForwardRef>
          <ForwardRef
            as="select"
            name="birthMonth"
            value="January"
          >
            <option
              selected={true}
              value=""
            >
              N/A
            </option>
            <option
              value="January"
            >
              January
            </option>
            <option
              value="February"
            >
              February
            </option>
            <option
              value="March"
            >
              March
            </option>
            <option
              value="April"
            >
              April
            </option>
            <option
              value="May"
            >
              May
            </option>
            <option
              value="June"
            >
              June
            </option>
            <option
              value="July"
            >
              July
            </option>
            <option
              value="August"
            >
              August
            </option>
            <option
              value="September"
            >
              September
            </option>
            <option
              value="October"
            >
              October
            </option>
            <option
              value="November"
            >
              November
            </option>
            <option
              value="December"
            >
              December
            </option>
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
      <ForwardRef
        className="paddingRight"
        lg={3}
        md={7}
      >
        <ForwardRef
          className="day-dropdown"
          controlId="formBirthday"
        >
          <ForwardRef
            as="select"
            name="birthday"
            value="1"
          >
            <option
              selected={true}
              value=""
            >
              N/A
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
            <option
              value="24"
            >
              24
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="26"
            >
              26
            </option>
            <option
              value="27"
            >
              27
            </option>
            <option
              value="28"
            >
              28
            </option>
            <option
              value="29"
            >
              29
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="31"
            >
              31
            </option>
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
      <ForwardRef
        lg={3}
        md={7}
      >
        <ForwardRef
          controlId="formAnniversayMonth"
        >
          <ForwardRef
            className="font-weight-bold text-condensed font-size-base"
            column={false}
            srOnly={false}
          >
            Work Anniversary
          </ForwardRef>
          <ForwardRef
            as="select"
            name="anniversaryMonth"
            value="December"
          >
            <option
              selected={true}
              value=""
            >
              N/A
            </option>
            <option
              value="January"
            >
              January
            </option>
            <option
              value="February"
            >
              February
            </option>
            <option
              value="March"
            >
              March
            </option>
            <option
              value="April"
            >
              April
            </option>
            <option
              value="May"
            >
              May
            </option>
            <option
              value="June"
            >
              June
            </option>
            <option
              value="July"
            >
              July
            </option>
            <option
              value="August"
            >
              August
            </option>
            <option
              value="September"
            >
              September
            </option>
            <option
              value="October"
            >
              October
            </option>
            <option
              value="November"
            >
              November
            </option>
            <option
              value="December"
            >
              December
            </option>
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
      <ForwardRef
        lg={3}
        md={5}
      >
        <ForwardRef
          className="day-dropdown"
          controlId="formAnniversayday"
        >
          <ForwardRef
            as="select"
            name="anniversaryDay"
            value="11"
          >
            <option
              selected={true}
              value=""
            >
              N/A
            </option>
            <option
              value="1"
            >
              1
            </option>
            <option
              value="2"
            >
              2
            </option>
            <option
              value="3"
            >
              3
            </option>
            <option
              value="4"
            >
              4
            </option>
            <option
              value="5"
            >
              5
            </option>
            <option
              value="6"
            >
              6
            </option>
            <option
              value="7"
            >
              7
            </option>
            <option
              value="8"
            >
              8
            </option>
            <option
              value="9"
            >
              9
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
            <option
              value="13"
            >
              13
            </option>
            <option
              value="14"
            >
              14
            </option>
            <option
              value="15"
            >
              15
            </option>
            <option
              value="16"
            >
              16
            </option>
            <option
              value="17"
            >
              17
            </option>
            <option
              value="18"
            >
              18
            </option>
            <option
              value="19"
            >
              19
            </option>
            <option
              value="20"
            >
              20
            </option>
            <option
              value="21"
            >
              21
            </option>
            <option
              value="22"
            >
              22
            </option>
            <option
              value="23"
            >
              23
            </option>
            <option
              value="24"
            >
              24
            </option>
            <option
              value="25"
            >
              25
            </option>
            <option
              value="26"
            >
              26
            </option>
            <option
              value="27"
            >
              27
            </option>
            <option
              value="28"
            >
              28
            </option>
            <option
              value="29"
            >
              29
            </option>
            <option
              value="30"
            >
              30
            </option>
            <option
              value="31"
            >
              31
            </option>
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <ForwardRef>
      <ForwardRef>
        <ForwardRef
          className="pr-md-2"
          controlId="formLanguagesSpoken"
        >
          <Unknown
            allowCheckbox={false}
            className="font-weight-bold text-condensed font-size-base"
            controlId="formLanguagesSpoken"
            label="Languages Spoken"
            name="languagesSpoken"
            type="text"
            value="English, Spanish"
          />
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <Unknown
      className="font-weight-bold text-condensed font-size-base"
      controlId="formSkills"
      label="Professional Skills"
      name="professionalSkills"
      rows={3}
      value="Lorem Ipsum is simply dummy text of the printing and typesetting     industry. Lorem Ipsum has been the industry\\\\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries."
    />
    <Unknown
      className="font-weight-bold text-condensed font-size-base"
      controlId="formCurrentlyWorking"
      label="What I'm currently working on"
      name="currentlyWorking"
      rows={3}
      value="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
    />
    <Unknown
      className="font-weight-bold text-condensed font-size-base"
      controlId="formAboutme"
      label="Something about me"
      name="aboutMe"
      rows={3}
      value="Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
    />
    <ForwardRef
      className="mt-2 mb-2"
    >
      <ForwardRef
        className="d-flex justify-content-end mt-2"
        md={12}
      >
        <Unknown
          className="cancel-button text-underlined text-upper border border-0 mr-3"
          handleClick={[Function]}
          title="Cancel"
          uniqueIdentifier="ok"
        />
        <Unknown
          className="text-upper"
          handleClick={[Function]}
          title="Save"
          uniqueIdentifier="ok"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</Unknown>
`;

exports[`GetToKnowMeForm should render correctly 2`] = `
<Unknown
  subtitle="All information is visible to anyone on UMG Central."
  subtitleClass="gray-600 font-size-sm d-lg-inline ml-lg-4 form-user-desc-subtitle"
  title="Contact Me"
  titleClass="font-size-md font-weight-bold text-uppercase d-lg-inline flex-row mb-0"
  topBorder="darkTurquoise"
  wrapperClass="h-100"
>
  <ForwardRef
    className="form-user-description mt-4"
    inline={false}
  >
    <ForwardRef>
      <ForwardRef>
        <ForwardRef
          className="pr-md-2"
          controlId="formMobile"
        >
          <Unknown
            allowCheckbox={false}
            className="font-weight-bold text-condensed"
            controlId="formMobile"
            label="Mobile"
            name="mobileNumber"
            placeholder="+1 999 999 9999"
            type="tel"
            value="9900990099"
          />
        </ForwardRef>
      </ForwardRef>
    </ForwardRef>
    <ForwardRef
      className="mt-2 mb-2"
    >
      <ForwardRef
        className="d-flex justify-content-end mt-2"
        md={12}
      >
        <Unknown
          className="cancel-button text-underlined text-upper border border-0 mr-3"
          handleClick={[Function]}
          title="Cancel"
          uniqueIdentifier="ok"
        />
        <Unknown
          className="text-upper"
          handleClick={[Function]}
          title="Save"
          uniqueIdentifier="ok"
        />
      </ForwardRef>
    </ForwardRef>
  </ForwardRef>
</Unknown>
`;
